### PR TITLE
Currency API fixes

### PR DESCRIPTION
**Pull request by jeyvia**
- added rate to convert-currency command (optional field)
With currency API working: 
- If rate is provided, command will use provided rate
- If not, will use hashmap's rate
With currency API not working:
- Will use rate provided only
- If rate not provided, will throw exception

### CODE REVIEW

File: src/main/java/seedu/moneygowhere/data/currency/CurrencyManager.java [hunk 1]
```diff
@@ -25,10 +26,15 @@ public void addCurrency(Currency currency) {
...
-    public boolean hasCurrency(String currency) throws CurrencyInvalidException {
+    public boolean hasCurrency(String currency) throws CurrencyInvalidException, CurrencyRatesNotFoundException {
         if (exchangeRates.containsKey(currency)) {
             return true;
         }
+        if (exchangeRates.isEmpty()) {
+            System.out.println("no way");
+            throw new CurrencyRatesNotFoundException(Messages.CURRENCY_MANAGER_RATES_NOT_FOUND);
+        }
+        System.out.println("lolol");
```
Comment (xzynos) [inline src/main/java/seedu/moneygowhere/data/currency/CurrencyManager.java:37]: Haha may I check is this an intended error message?

**Review by xzynos — COMMENTED** [review]
The code looks good overall! Thanks for fixing the issue with the currency API.

File: src/main/java/seedu/moneygowhere/data/currency/CurrencyManager.java (hunk 1 above)
Comment (jeyvia) [reply src/main/java/seedu/moneygowhere/data/currency/CurrencyManager.java:37]: oh my i was doing some testing and i forgot to remove it. My bad!

Comment (xzynos) [reply src/main/java/seedu/moneygowhere/data/currency/CurrencyManager.java:37]: Hahahaha no problem. Glad we caught it before we merged.

**Review by xzynos — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**